### PR TITLE
Warn if git lfs not installed

### PR DESCRIPTION
**Pull request by g-arjones**

### CODE REVIEW

File: lib/autobuild/import/git-lfs.rb [hunk 1]
```diff
@@ -1,9 +1,13 @@
-# Disable git-lfs at checkout time, we run install --local and pull later 
+# Disable git-lfs at checkout time, we run install --local and pull later
 Autobuild::Git.default_config['filter.lfs.smudge'] = 'git-lfs smudge --skip -- %f'
 Autobuild::Git.default_config['filter.lfs.required'] = 'false'
 
 module Autobuild
     Git.add_post_hook(always: true) do |importer, package|
+        if importer.uses_lfs?(package) && !importer.lfs_installed?
```
[the remark this answers is not in the document]
Comment (g-arjones) [reply lib/autobuild/import/git-lfs.rb]: >  It detects whether it is in use with the presence of .git/lfs

The goal is to warn the user if the repository uses lfs AND git lfs is not installed. `.git/lfs` is present only if git lfs is already installed so it does not fit this use case.

> I would suggest making lfs_installed?

Good point.. :+1:

[the remark this answers is not in the document]
Comment (doudou) [reply lib/autobuild/import/git-lfs.rb]: > The goal is to warn the user if the repository uses lfs AND git lfs is not installed. .git/lfs is present only if git lfs is already installed so it does not fit this use case.

Oh. Algright. I thought that .git/lfs wouid be present no matter what.

File: lib/autobuild/import/git.rb [hunk 2]
```diff
@@ -1095,6 +1096,22 @@ def each_alternate_path(package)
...
         end
 
+        def uses_lfs?(package)
+            git_files = run_git(package, 'ls-files').join("\n")
+            git_attrs = run_git(
+                package, 'check-attr', '--all', '--stdin',
+                input_streams: [StringIO.new(git_files)]
+            ).join("\n")
+
+            !git_attrs.scan(/(.*): filter: lfs/).empty?
```
[the remark this answers is not in the document]
Comment (g-arjones) [reply lib/autobuild/import/git.rb]: :+1:

**Comment by g-arjones** [comment]
Don't merge it yet, there is a problem with this.. [#write](https://github.com/rock-core/autobuild/blob/0ed7e96126e5e33ba3223b1fa97b03aca762b7c6/lib/autobuild/subcommand.rb#L356) blocks forever if `instream` has too many lines (1000+). Apparently, the pipe gets filled but I don't understand why since the forked process is (should be?) reading it. 

Ideas?

**Comment by g-arjones** [comment]
Fixed it.. 

The pipe was getting full because the parent process only read its end of the pipe after it finished writing... I think we are good to go now.

**Pull request merged**